Assets/Scripts/Unit/Action/Signal/SignalUI.cs

```diff
@@ -0,0 +1,143 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class SignalUI : ActionUI
+{
+    Tile target;
+    Signal s;
+
+    void Start()
```

Comment:
Have you tried running the skill multiple times? I think Start only runs once in the GameObject's lifecycle. Wouldn't target not be set correctly?

Reply:
Alternative: Override `Initialize` method (don't forget to call base.Initialize())